dbms/src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -308,7 +323,15 @@ class AggregateFunctionSum final : public IAggregateFunctionDataHelper<Data, Agg
 
     AggregateFunctionSum(PrecType prec, ScaleType scale)
     {
-        std::tie(result_prec, result_scale) = SumDecimalInferer::infer(prec, scale);
+        if (Name::name == NameFinalSumStage::name)
+        {
+            result_prec = prec;
+            result_scale = scale;
+        }
+        else
+        {
+            std::tie(result_prec, result_scale) = SumDecimalInferer::infer(prec, scale);
+        }
```

Comment:
> Looks like it could be made into a template method.
> 
> ```
> struct NameFinalSumStage
> {
>     static constexpr auto name = "finalSumStage";
> 
>     static std::pair<PrecType, ScaleType> decimalInfer(PrecType prec, ScaleType scale)
>     {
>         return {prec, scale};
>     }
> };
> 
> struct NameSumKahan
> {
>     static constexpr auto name = "sumKahan";
> 
>     static std::pair<PrecType, ScaleType> decimalInfer(PrecType prec, ScaleType scale)
>     {
>         return SumDecimalInferer::infer(prec, scale);
>     }
> };
> 
> std::tie(result_prec, result_scale) = Name::decimalInfer(prec, scale);
> ```

It's a great idea!